loader/loader.go

```diff
@@ -1188,110 +1188,367 @@ func fetchMatchedLiteral(ctx *tcontext.Context, router *router.Table, schema, ta
 	return targetSchema, targetTable
 }
 
-func (l *Loader) restoreData(ctx context.Context) error {
-	begin := time.Now()
+// restore schema job
+type restoreSchemaJob struct {
+	loader           *Loader
+	session          *DBConn
+	database         string
+	table            string // empty if it's a schema of database
+	pathOfSchemaFile string
+}
 
-	baseConn, err := l.toDB.GetBaseConn(ctx)
-	if err != nil {
-		return err
+// check is schema of database
+func (job *restoreSchemaJob) isSchemaOfDatabase() (yes bool) {
+	yes = (job.table == "")
+	return
+}
+
+// check is schema of table
+func (job *restoreSchemaJob) isSchemaOfTable() (yes bool) {
```

Comment:
```suggestion
func (job *restoreSchemaJob) isTable() (yes bool) {
```

Reply:
isTable is more concise, isSchemaOfTable may be more explicit?

Reply:
I prefer isDB / isTable, because the receiver is of type `restoreSchemaJob`, developer should know that's dealing with schema

Reply:
Ok, it's acceptable to me both. xD